pkg/controller/direct/common/normalize.go

```diff
@@ -0,0 +1,52 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package common
+
+import (
+	"time"
+
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+func NormalizeStringToTimestamp(ts string) *timestamppb.Timestamp {
```

Comment:
This seems _very_ permissive.... did we previously support all this formats?

Reply:
The GCP timestamp proto does. This is for comparison. I feel like some previous Update compare could be wrong.. (like the same timestamp meaning with a different format is considered different)

Reply:
I changed this in #2904 to make it less permissive.